test/load/terraform/test-servers.tf

```diff
@@ -0,0 +1,86 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# Start up test servers to run tests from
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_security_group" "test-servers" {
+  name   = "test-server-sg"
+  vpc_id = module.vpc.vpc_id
+
+  ingress {
+    from_port       = 8500
+    to_port         = 8500
+    security_groups = [module.consul.security_group_id_clients]
+    protocol        = "6"
+    cidr_blocks     = ["0.0.0.0/0"]
+  }
+  ingress {
+    from_port   = 5557
+    to_port     = 5558
+    protocol    = "6"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "6"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_instance" "test-server-primary" {
+  ami                         = var.test_server_ami
+  instance_type               = var.test_instance_type
+  key_name                    = module.keys.key_name
+  subnet_id                   = module.vpc.public_subnets[0]
+  security_groups             = [aws_security_group.test-servers.id]
+  associate_public_ip_address = var.test_public_ip
+  tags = {
+    Name = "test-server-primary"
+  }
+  user_data = templatefile(
+    "./start-locust-primary.sh",
+    {
+      lb_endpoint = module.alb.this_lb_dns_name
+    }
+  )
+}
+
+resource "aws_launch_configuration" "test-servers" {
+  name_prefix     = "${var.cluster_name}-test-"
+  image_id        = var.test_server_ami
+  instance_type   = var.test_instance_type
+  key_name        = module.keys.key_name
+  security_groups = [aws_security_group.test-servers.id]
+
+  associate_public_ip_address = var.test_public_ip
+  lifecycle {
+    create_before_destroy = true
+  }
+  user_data = templatefile(
+    "./start-locust-worker.sh",
+    {
+      primary_ip  = aws_instance.test-server-primary.public_ip
+      lb_endpoint = module.alb.this_lb_dns_name
+    }
+  )
+}
+
+resource "aws_autoscaling_group" "test-servers" {
+  name                      = aws_launch_configuration.test-servers.name
+  launch_configuration      = aws_launch_configuration.test-servers.id
+  min_size                  = 2
+  max_size                  = 5
+  desired_capacity          = 2
+  wait_for_capacity_timeout = "480s"
```

Comment:
It normally takes this long to spin up?

Reply:
No, that's the timeout in case it decides to take 10 minutes to spin up. I am just running off the [defaults defined](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#wait_for_capacity_timeout). I can change it though!

Reply:
Ah ok, the default said 10m and this is 8m (if my math is right) so I thought the arbitrary value may be important for some reason here.

Reply:
Oh I can't do math, my b. We can change if necessary?